console/v1/types_console_quick_start.go

```diff
@@ -0,0 +1,93 @@
+package v1
+
+import (
+	authorizationv1 "openshift/api/pgk/authorization/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ConsoleQuickStart is an extension for guiding user thought various
+// workflows in the OpenShift web console.
+type ConsoleQuickStart struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ConsoleQuickStartSpec `json:"spec"`
+}
+
+// ConsoleQuickStartSpec is the desired quick start configuration.
+type ConsoleQuickStartSpec struct {
+	// displayName is the display name of the Quick Start.
+	DisplayName string `json:"displayName"`
+	// icon is a base64 encoded image that are used as Quick Start's
+	// +optional
+	Icon string `json:"icon,omitempty"`
+	// tags is an array of strings that describe the Quick Start.
+	// +optional
+	Tags []string `json:"tags,omitempty"`
+	// duration informs user on aproximately how long the Quick Start should take to finish.
+	// +optional
+	Duration int `json:"duration,omitempty"`
+	// description is the description of the Quick Start. (includes markdown)
+	// +optional
+	Description string `json:"description,omitempty"`
+	// prerequisites contains a list prerequisites for the Quick.
+	// +optional
+	Prerequisites []string `json:"prerequisites,omitempty"`
+	// introduction describes the purpose and usability of the Quick Start. (includes markdown)
+	Introduction string `json:"introduction"`
+	// tasks contains list of tasks that user has to take in order to pass the Quick Start.
+	Tasks []ConsoleQuickStartTask `json:"tasks"`
+	// conclusion sums up the Quick Start and suggests the possible
+	// next Quick Starts for user to take. (includes markdown)
+	// +optional
+	Conclusion string `json:"conclusion,omitempty"`
+	// nextQuickStart is the name of the following Quick Start.
+	// +optional
+	NextQuickStart string `json:"nextQuickStart,omitempty"`
```

Comment:
Should this be an array in case there's ever more than one suggestion?

Reply:
I'm thinking about the use case where you install an operator first, then there are a couple different sample apps you could deploy with different quick starts. I don't feel strongly about this, but I'm asking now since it's hard to change to an array later.

Reply:
The designs indicated a single value. But having as an array does make more sense. Perhaps we should bring this up with UX again.

Reply:
@spadgett makes sense will update.